.github/workflows/ci.yml

```diff
@@ -79,11 +80,29 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # checkout all commits to be able to determine merge base
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - name: Fetch base ref to find merge-base for GIB
+        run: .github/bin/git-fetch-base-ref.sh
       - name: Check Commits
         uses: trinodb/github-actions/block-commits@c2991972560c5219d9ae5fb68c0c9d687ffcdd10
         with:
           action-merge: fail
           action-fixup: none
+      - name: Check if all intermediate commits of this PR compile
+        run: |
+          git config user.name "$(git --no-pager log --format=format:'%an' -n 1)"
+          git config user.email "$(git --no-pager log --format=format:'%ae' -n 1)"
+          git log --graph --oneline --all
```

Comment:
list of all commits, ever? this is probably long

Reply:
Yes, this is only temporary, but I need to see what state GitHub leaves the local repo in.
I'm trying to figure out how to only compile intermediate commits, since the HEAD of the PR is already compiled and tested.

Reply:
I'll print the git tree in the PR, but limit the range to the PR branch

Reply:
After looking at results from running jobs, it looks like printing a constant number of commits before the merge base is a bad idea - if a PR branch was forked a long time ago, it won't print the fork point and it's difficult to see both sides of the merge. I'll change this so that both merge branches are visible.